data/interfaces/default/js/tables/history_table.js

```diff
@@ -231,24 +239,58 @@ history_table_options = {
         });
 
         if ($('#row-edit-mode').hasClass('active')) {
-            $('.delete-control').each(function() {
+            $('.delete-control').each(function () {
                 $(this).removeClass('hidden');
             });
         }
+
+        history_table.rows().every(function () {
+            var rowData = this.data();
+            if (rowData['group_count'] != 1 && rowData['reference_id'] in history_child_table) {
+                // if grouped row and a child table was already created
+                $(this.node()).find('i.fa').toggleClass('fa-plus-circle').toggleClass('fa-minus-circle');
+                this.child(childTableFormat(rowData)).show();
+                createChildTable(this, rowData)
+            }
+        });
     },
     "preDrawCallback": function(settings) {
         var msg = "<i class='fa fa-refresh fa-spin'></i>&nbspFetching rows...";
         showMsg(msg, false, false, 0)
     },
-    "rowCallback": function (row, rowData) {
-        if ($.inArray(rowData['id'], history_to_delete) !== -1) {
+    "rowCallback": function (row, rowData, rowIndex) {
+        if (rowData['group_count'] == 1) {
+            // if no grouped rows simply toggle the delete button
+            if ($.inArray(rowData['id'], history_to_delete) !== -1) {
+                $(row).find('button[data-id="' + rowData['id'] + '"]').toggleClass('btn-warning').toggleClass('btn-danger');
+            }
+        } else {
+            // if grouped rows
+            // toggle the parent button to danger
             $(row).find('button[data-id="' + rowData['id'] + '"]').toggleClass('btn-warning').toggleClass('btn-danger');
+            // check if any child rows are not selected
+            for (var i = rowData['reference_id']; i <= rowData['id']; i++) {
```

Comment:
I think I have an idea. I can use `GROUP_CONCAT` in SQLite to get a comma delimited string of individual row `id`'s associated with the `referencd_id`, then use that list instead of the for loop.